todos-auth-fluent/Sources/App/Models/User.swift

```diff
@@ -0,0 +1,111 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2021-2021 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import FluentKit
+import Foundation
+import Hummingbird
+import HummingbirdAuth
+
+/// Database description of a user
+final class User: Model, HBAuthenticatable {
+    static let schema = "user"
+
+    @ID(key: .id)
+    var id: UUID?
+
+    @Field(key: "email")
+    var email: String
+
+    @Field(key: "name")
+    var name: String
+
+    @Field(key: "password")
+    var passwordHash: String
+
+    @Children(for: \.$owner)
+    var todos: [Todo]
+
+    internal init() {}
+
+    internal init(id: UUID? = nil, name: String, email: String, passwordHash: String) {
+        self.id = id
+        self.name = name
+        self.email = email
+        self.passwordHash = passwordHash
+    }
+
+    internal init(from userRequest: CreateUserRequest) {
+        self.id = nil
+        self.name = userRequest.name
+        self.passwordHash = Bcrypt.hash(userRequest.password, cost: 12)
+    }
+}
+
+extension User {
+    /// create a User in the db attached to request
+    static func create(name: String, email: String, password: String, request: HBRequest) async throws -> User {
+        // check if user exists and if they don't then add new user
+        let existingUser = try await User.query(on: request.db)
+            .filter(\.$email == email)
+            .first()
+        // if user already exist throw conflict
+        guard existingUser == nil else { throw HBHTTPError(.conflict) }
+
+        // Encrypt password on a separate thread
+        let passwordHash = try await Bcrypt.hash(password, cost: 12, for: request).get()
```

Comment:
We should provide a `.default` for the cost, or at least clarify what it means. Because people who 'like performance' might tune this down to low numbers like 6 and not know what they did.

Reply:
That would be a big problem for password security

Reply:
That would have to be in HummingbirdAuth where those changes would go, not in the sample

Reply:
That's fair!

Reply:
Would you be able to add a todo, or even a PR